packages/artifact/src/internal-download-http-client.ts

```diff
@@ -12,122 +12,174 @@ import {
   QueryArtifactResponse
 } from './internal-contracts'
 import {IHttpClientResponse} from '@actions/http-client/interfaces'
-import {HttpClient} from '@actions/http-client'
+import {HttpManager} from './internal-http-manager'
```

Comment:
It seems everything is `internal` 😄 , and naming every file with prefix `internal` adds clutter.  We could either create a folder `internal` and move all files there, or name the only `artifact-client` file `external-artifact-client`.   Then we can drop the `internal` prefix from everywhere.

Reply:
The original motivation behind this is the `glob` package: https://github.com/actions/toolkit/tree/master/packages/glob/src

I've been following the structure pretty closely and `internal` is used to clearly differentiate between what is public and what isn't. There are a lot of internal methods like `listArtifact` or `patchSize` that I don't want people poking around with and this helps set a clear boundary.

Given the large amount of files though, I'll probably move them into a separate directory.

Reply:
I've updated the directory structure, previously `internal` files now are in a separate directory
 
<img src="https://user-images.githubusercontent.com/16109154/76430178-4f0f0600-63b0-11ea-882d-2a148639a0a0.png" width="150" height="300" />